frontend/catalyst/pennylane_extensions.py

```diff
@@ -323,7 +323,7 @@ def default_expand_fn(self, circuit, max_expansion=10):
         decompose_to_qubit_unitary = QJITDevice._get_operations_to_convert_to_matrix(self.config)
 
         def _decomp_to_unitary(self, *_args, **_kwargs):
-            return [qml.QubitUnitary(qml.matrix(self), wires=self.wires)]
+            return [qml.QubitUnitary(self.matrix(), wires=self.wires)]
```

Comment:
It's good to add a test that shows what this change actually fixes for the future references.

Reply:
(see my comment 😅 )

Reply:
didn't see that in my first pass :smile:

Reply:
https://github.com/PennyLaneAI/catalyst/pull/468/commits/8dc026a35cd2ec908c02038f852d8f6dab0e5b70